build.rs

```diff
@@ -206,3 +214,50 @@ impl PropertyRule {
         format!("super::PropertyRule {{ required: {required}, type_rule: super::TypeRule::{type_rule} }}")
     }
 }
+
+fn zip_test_snapshots() {
+    // Zip the expected output files for the end-to-end tests so that they can be included in the test binary. This will not affect the size of the cdk-from-cfn binary.
+
+    let src_dir = "./tests/end-to-end";
+    let dst_file = "./tests/end-to-end-test-snapshots.zip";
+    let do_not_zip = ["app-boiler-plate-files", "working-dir"];
+
+    let file = fs::File::create(Path::new(dst_file)).unwrap();
+
+    let walkdir = WalkDir::new(src_dir);
+    let mut zip = ZipWriter::new(file);
+    let options = FileOptions::default();
+    let mut buffer = Vec::new();
+
+    'dir_entries: for entry in walkdir.into_iter().map(|e| e.unwrap()) {
+        let path = entry.path();
+        let name = path
+            .strip_prefix(Path::new(src_dir))
+            .expect(&format!("{src_dir} should be a prefix of {:?}", path))
+            .to_str()
+            .expect("failed to convert filename to string");
+
+        for d in do_not_zip {
+            if name.contains(d) {
+                continue 'dir_entries;
```

Comment:
do we need this label?

Reply:
yes, because at this point I want to continue on the outer loop, not this inner one.